pkg/cmd/server/handlers/authentication.go

```diff
@@ -1,34 +0,0 @@
-package handlers
-
-import (
-	"net/http"
-
-	"github.com/golang/glog"
-
-	"k8s.io/apiserver/pkg/authentication/authenticator"
-	apirequest "k8s.io/apiserver/pkg/endpoints/request"
-)
-
-// AuthenticationHandlerFilter creates a filter object that will enforce authentication directly
-func AuthenticationHandlerFilter(handler http.Handler, authenticator authenticator.Request, contextMapper apirequest.RequestContextMapper) http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		user, ok, err := authenticator.AuthenticateRequest(req)
-		if err != nil || !ok {
-			http.Error(w, "Unauthorized", http.StatusUnauthorized)
-			return
-		}
-
-		ctx, ok := contextMapper.Get(req)
-		if !ok {
-			http.Error(w, "Unable to find request context", http.StatusInternalServerError)
-			return
-		}
-		if err := contextMapper.Update(req, apirequest.WithUser(ctx, user)); err != nil {
```

Comment:
Upstream does not seem to have these error checks.  I assume we OK with that?

Reply:
Upstream also seems to gather Prometheus metrics per user?

`Counter of authenticated requests broken out by username.`

Reply:
> Upstream does not seem to have these error checks. I assume we OK with that?

Yes, though I wouldn't mind if you wanted to add them.



> Upstream also seems to gather Prometheus metrics per user?
> 
> Counter of authenticated requests broken out by username.

I guess we'll see how that works here.  If it needs fixing, we'll should fix upstream and pick down.

Reply:
@smarterclayton has mentioned the need to avoid high cardinality data with Prometheus.  This seems like it could cause issues.